### ISSUE TITLE
Incorrect font rendering + assertion failure

### ISSUE THREAD

**Issue by torokati44** [ruffle-rs/ruffle#9477]
### Describe what you were doing

I was trying to create a minimal reproducer (with JPEXS ffdec) for an unrelated issue with font rendering, and instead got this panic...
Here is the file in question:  [fonttest.zip](https://github.com/ruffle-rs/ruffle/files/10690904/fonttest.zip)

The font was extracted from Robot Unicorn Attack, where it is also drawn incorrectly.

FP on the left, (crashed) Ruffle on the right:
![image](https://user-images.githubusercontent.com/288816/217676682-82f7fba5-374f-488d-b93b-2f29d8b97f17.png)

So this is a 2 in 1 bug report, really unprofessional, sorry about it... :|

### What does the crash message say?

panicked at 'assertion failed: `(left == right)`
  left: `Some(10253)`,
 right: `None`: [] Gotos must start from the correct tag position for frame 2', core/src/display_object/movie_clip.rs:1536:9

### Ruffle Version

0.1.0-nightly (aa7c0a5d0 2023-01-25)

### Affected platform

Desktop app

### Operating system

Fedora 37 (ThirtySeven) [64-bit]

### Additional information

Filename: fonttest.swf

### Render Info
Renderer: wgpu
Adapter Backend: Vulkan
Adapter Name: "AMD Radeon RX 6600 XT (RADV NAVI23)"
Adapter Device Type: DiscreteGpu
Adapter Driver Name: "radv"
Adapter Driver Info: "Mesa 22.3.4"
Enabled features: PUSH_CONSTANTS
Available features: DEPTH_CLIP_CONTROL | DEPTH32FLOAT_STENCIL8 | TEXTURE_COMPRESSION_BC | INDIRECT_FIRST_INSTANCE | TIMESTAMP_QUERY | PIPELINE_STATISTICS_QUERY | SHADER_FLOAT16 | MAPPABLE_PRIMARY_BUFFERS | TEXTURE_BINDING_ARRAY | BUFFER_BINDING_ARRAY | STORAGE_RESOURCE_BINDING_ARRAY | SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING | UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING | PARTIALLY_BOUND_BINDING_ARRAY | MULTI_DRAW_INDIRECT | MULTI_DRAW_INDIRECT_COUNT | ADDRESS_MODE_CLAMP_TO_BORDER | POLYGON_MODE_LINE | POLYGON_MODE_POINT | TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | SHADER_FLOAT64 | CONSERVATIVE_RASTERIZATION | VERTEX_WRITABLE_STORAGE | CLEAR_TEXTURE | SPIRV_SHADER_PASSTHROUGH | SHADER_PRIMITIVE_INDEX | MULTIVIEW | TEXTURE_FORMAT_16BIT_NORM | ADDRESS_MODE_CLAMP_TO_ZERO | WRITE_TIMESTAMP_INSIDE_PASSES
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 8388606, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 2147483648, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 32, min_storage_buffer_offset_alignment: 32, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface samples: 4
Surface size: Extent3d { width: 800, height: 600, depth_or_array_layers: 1 }

**Comment by torokati44** [comment]
Changing the `FillOptions::even_odd()` parameter of `tessellate_path` to `FillOptions::non_zero()` in `tessellator.rs` fixes this particular issue, but it may (likely) break some other things...

**Comment by n0samu** [comment]
I think you made a mistake with creating this SWF in JPEXS. Actually the debug Flash projector does not accept this SWF either. (Although I guess the standard projector is a bit more lenient and still works)

**Comment by torokati44** [comment]
That's entirely possible, yes, if not highly likely even.
Although, an argument can be made that even though it's malformed, if the regular FP accepts it, perhaps Ruffle should too? It might not even be the first such thing.

Anyway, I'm more concerned at the moment with the rendering part, the panic is just an unfortunate side effect along the way. If someone could fix up the swf to make it no longer panic Ruffle, that would be great!

**Comment by n0samu** [comment]
I'm a bit confused because I cannot reproduce the crash even with the exact same Ruffle version that you're using. (I'm on Windows though, not Linux.)

Can you check whether this SWF fixes the crash for you? You made the same mistake that I often make when creating SWFs in JPEXS, which is that you forgot to add a ShowFrame tag at the end of your frame. [fonttest.zip](https://github.com/ruffle-rs/ruffle/files/10691070/fonttest.zip)

**Comment by torokati44** [comment]
Huhh, that's... weird. I can't test right now, sorry, just got off the PC. I'll try tomorrow.

**Comment by n0samu** [comment]
Alright, I found the error message from the panic:

https://github.com/ruffle-rs/ruffle/blob/83f88a6f3e6757993d09b6bc97303739f86d4c30/core/src/display_object/movie_clip.rs#L1550

Looks like this assert is only enabled with the `timeline_debug` feature. So that's why it's not panicking for me. I tried building with `timeline_debug` and now I'm seeing the panic too.

And with the ShowFrame tag added, I can confirm that the SWF no longer panics.
